modules/internal/DefaultSparse.chpl

```diff
@@ -235,7 +235,114 @@ module DefaultSparse {
         add_help(ind);
       }
     }
-  
+
+    proc bulkAdd(inds: [] index(rank, idxType), sorted=false, noDuplicate=false){
+      var numAdded = inds.size; //maybe remove this var
+
+      //find individual insert points
+      //and eliminate duplicates between inds and dom
+      var indivInsertPts: [{0..#numAdded}] int;
+      var actualInsertPts: [{0..#numAdded}] int; //where to put in newdom
+
+      if !sorted then QuickSort(inds);
```

Comment:
It seems surpsising to me that the passed array will be modified. E.g. if I do

      someDomain += myArray;

it's not immediately obvious when looking that myArray will be modified.
It could be documented well, but I think it's probably better to sort the
array in temporary space. Another interesting possibility would be to
allow bulkAdd to use an argument indicating if inds can be written to.
But, if += is not really useful if the array is copied and then sorted...
then we could talk more about how to make this clearer.